frontend/www/js/omegaup/contest/edit.ts

```diff
@@ -113,23 +114,51 @@ OmegaUp.on('ready', () => {
           problems: this.problems,
           requests: this.requests,
           users: this.users,
-          existingProblems: this.existingProblems,
+          searchResultProblems: this.searchResultProblems,
+          searchResultUsers: this.searchResultUsers,
         },
         on: {
-          'update-existing-problems': (query: string) => {
+          'update-search-result-problems': (query: string) => {
             api.Problem.list({
               query,
             })
               .then((data) => {
-                this.existingProblems = [];
-                data.results.forEach((problem: types.ProblemListItem) => {
-                  this.existingProblems.push({
-                    key: problem.alias,
-                    value: problem.title,
+                // Problems previously added into the contest should not be
+                // shown in the dropdown
+                const addedProblems = new Set(
+                  this.problems.map((problem) => problem.alias),
+                );
+                const searchedProblems = new Set(
+                  data.results.map((problem) => problem.alias),
+                );
+                this.searchResultProblems = [...searchedProblems]
+                  .filter((problem) => !addedProblems.has(problem))
+                  .map((key) => ({ key, value: key }));
+              })
+              .catch(ui.apiError);
+          },
+          'update-search-result-users': (query: string) => {
+            api.User.list({ query })
+              .then((data) => {
+                // Users previously invited to the contest should not be shown
+                // in the dropdown
+                const addedUsers = new Set(
+                  this.users.map((user) => user.username),
+                );
+                const searchedUsers = new Set(data.map((user) => user.label));
+                const algo = [...searchedUsers]
```

Comment:
```suggestion
                this.searchResultUsers = data
```

mismo comentario que arriba.